docs/software/devtools/debug/linaro.md

```diff
@@ -0,0 +1,294 @@
+> uenv view default
+> nvcc -c -arch=sm_90 -g -G test_gpu.cu
+> mpicxx -g test_cpu.cpp test_gpu.o -o myexe
+```
+
+### Launch Linaro DDT
+
+To use the DDT client with uenv, it must be launched in `Manual Launch` mode
+(assuming that it is connected to Alps via `Remote Launch`):
+
+=== "on local machine"
+
+    Start DDT, and connect to the target cluster using the drop down menu for Remote Launch.
+
+    Click on Manual launch, set the number of processes to listen to, 
+    then wait for the slurm job to start (see the "on Alps" tab).
+        
+    <img src="https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/0.png" width="600" />
+
+=== "on Alps"
+
+    log into the system and launch with the srun command:
+
+    ```bash
+    # start a session with both the PE used to build your application
+    # and the linaro-forge uenv mounted
+    > uenv start prgenv-gnu/24.11:v1,linaro-forge/24.1.1:v1 --view=prgenv-gnu:default
+    > source /user-tools/activate
+
+    > srun -N1 -n4 -t15 -pdebug \
+        ./cuda_visible_devices.sh   ddt-client   ./myexe
+    ```
+
+### Start debugging
+
+By default, DDT will pause execution on the call to MPI_Init:
+<img src="https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/1.png" width="600" />
+
+There are more than 1 mechanism for controlling program execution:
+
+=== "Breakpoint"
+
+    Breakpoint(s) can be set by clicking in the margin to the left of the line number:
+
+    <img src="https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/3.png" width="600" />
+
+=== "Stop at"
+
+    Execution can be paused in every CUDA kernel launch by activating the default breakpoints from the Control menu:
+
+    <img src="https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/4.png" width="400" />
+
+
+This screenshot shows a debugging session on 128 gpus:
+![DDTgpus](https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/5.png)
+
+More informations regarding how to use Linaro DDT are provided in the Forge
+[User Guide](https://docs.linaroforge.com/latest/html/forge/index.html).
+
+## Troubleshooting
+
+Notes on using specific systems:
+
+=== "alps"
+
+    !!! warning
+
+        Some clusters are not connected directly to the internet, hence some
+        environment variables need to be set so that the tool can connect to
+        the license server.
+
+        ```bash
+        export https_proxy=proxy.cscs.ch:8080
+        export http_proxy=proxy.cscs.ch:8080
+        export no_proxy=".local, .cscs.ch, localhost, 148.187.0.0/16, 10.0.0.0/8, 172.16.0.0/12"
+        ```
+
+        ???- note "default value of `http_proxy`"
+
+            By default the `https_proxy` and `http_proxy` variables are set to
+            `http://proxy.cscs.ch:8080`, as the transport is required for some
+            other services to work. You will have to set them for a debugging
+            session.
```

Comment:
Link to https://eth-cscs.github.io/cscs-docs/guides/internet-access/ instead? Since we only have Alps now, perhaps not necessary to call out Alps specifically?